pkg/pillar/cmd/volumemgr/volumemgr.go

```diff
@@ -237,9 +242,23 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	}
 	log.Functionf("user containerd ready")
 
+	// wait for kubernetes up if in kube mode, if gets error, move on
+	if ctx.hvTypeKube {
+		log.Noticef("volumemgr run: wait for kubernetes")
+		err := kubeapi.WaitForKubernetes(agentName, ps, stillRunning)
+		if err != nil {
+			log.Errorf("volumemgr run: wait for kubernetes error %v", err)
+		} else {
+			log.Noticef("volumemgr run: kubernetes node ready, longhorn ready")
+		}
+
+	}
+
 	if ctx.persistType == types.PersistZFS {
 		if isZvol, _ := zfs.IsDatasetTypeZvol(types.SealedDataset); isZvol {
+			// This code is called only on kubevirt eve
```

Comment:
But this code is not under
```
if ctx.hvTypeKube {
```
How are we sure, it's only kubevirt-related?

Reply:
@andrewd-zededa can answer that. I cross checked with him. Basically it means /persist/vault/ which is made out of zvol. That is only possible in Kubevirt eve case. In regular ZFS installs /persist/vault is a filesystem.

Reply:
But shouldn't we check a consistent predicate for that? Sometimes checking hvType and somethings checking the dataset doesn't make the code easier to understand hence might introduce bugs down the road.

Reply:
Actually I will let @andrewd-zededa comment on that. That code already exists I just added a comment to be clear.

Reply:
This question is still open.

Reply:
That code is nothing to do with this commit. I just added comment for the clarity.  I believe Andrew actually started with the hvType and he was made to change that in the reviews. @andrewd-zededa can you comment on that.